modules/_labs/slider/react/lib/Slider.tsx

```diff
@@ -0,0 +1,243 @@
+  },
+});
+
+const ProgressBarContainer = styled('div')({
+  position: 'absolute',
+  height: sliderTrackHeight,
+  top: '50%',
+  width: '100%',
+  transform: 'translateY(-50%)',
+});
+
+const ProgressBar = styled('progress')({
+  appearance: 'none',
+  width: sliderContainerWidth,
+  minWidth: sliderContainerMinWidth,
+  height: sliderTrackHeight,
+  overflow: 'hidden',
+  display: 'block',
+  border: 'none',
+  borderRadius: sliderTrackRadius,
+  '::-moz-progress-bar': {
+    background: colors.blueberry400,
+    borderRadius: sliderTrackRadius,
+  },
+  '::-webkit-progress-value': {
+    background: colors.blueberry400,
+    borderRadius: sliderTrackRadius,
+    transition: 'none',
+  },
+  '::-webkit-progress-bar': {
+    background: colors.soap300,
+  },
+  '::-ms-fill': {
+    border: 'none',
+    background: colors.blueberry400,
+    borderRadius: sliderTrackRadius,
+  },
+});
+
+const Wrapper = styled('div')({
+  display: 'flex',
+  flexDirection: 'row',
+  alignItems: 'center',
+});
+
+const InputValueBox = styled('input')({
+  ...type.body,
+  width: '48px',
+  height: spacing.xl,
+  border: `1px solid ${colors.licorice100}`,
+  borderRadius: borderRadius.m,
+  textIndent: spacing.xxxs,
+});
+
+/**
+ * This function is mainly used to find the percent fill value for the background gradients.
+ * Without it, the gradients pass the the dragger on tails ends every so slightly and is slightly visible in some browsers.
+ */
+const valueToPercent = (max: number, min: number, value: number): number => {
+  const percent = ((value - min) * 100) / (max - min);
+  const minPercent = 0;
+  const maxPercent = 100;
+  const outMin = 1;
+  const outMax = 99;
+  return ((percent - minPercent) * (outMax - outMin)) / (maxPercent - minPercent) + outMin;
+};
+
+export const Slider: React.FC<SliderProps> = ({
+  id,
+  max,
+  min,
+  step,
+  showTextInput,
+  onDragStart,
+  onEndDrag,
+  onChange,
+  startValue,
+}) => {
+  const [value, setValue] = React.useState(startValue);
+  const inputRef = React.useRef<HTMLInputElement>(null);
+
+  const onChangeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = Number(event.target.value);
+    setValue(value);
+    if (onChange) {
+      onChange(value);
+    }
+  };
+
+  const onSliderDragStart = () => {
+    if (onDragStart) {
+      onDragStart(value);
+    }
+  };
+
+  const onSliderDragEnd = () => {
+    if (onEndDrag) {
+      onEndDrag(value);
+    }
+  };
+
+  return (
+    <Wrapper>
+      <Interval>{min}</Interval>
+      <Container>
+        <ProgressBarContainer>
+          <ProgressBar value={valueToPercent(max, min, value)} max={100} />
+        </ProgressBarContainer>
+        <SliderInput
+          id={id}
+          type="range"
+          ref={inputRef}
+          max={max}
+          min={min}
+          step={step}
+          value={value}
+          onChange={onChangeHandler}
+          onMouseDown={onSliderDragStart}
+          onMouseUp={onSliderDragEnd}
```

Comment:
I think there are a couple aria attributes that can improve the accessibility of this:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_slider_role

Reply:
Are you talking about these?:
* aria-valuemin
* aria-valuemax
* aria-valuenow

`role=slider` is a superclass of `input[type=range]`. Do you know if these attributes are redundant given the `input` is a range already?

Reply:
A quick test with Voice Over seems like they are redundant, but might want to double check with Sam for other systems